File: frontend/src/types/imports.ts
export type ImportEntity = 'volunteers' | 'beneficiaries';

export interface ImportRowIssue {
  /** CSV file line number (header is line 1). */
  row: number;
  message: string;
}

export interface ImportReport {
  ok: boolean;
  total_rows: number;
  imported: number;
  skipped: ImportRowIssue[];
  errors: ImportRowIssue[];
}

File: backend/app/modules/pi/schemas/imports.py
"""CSV import schemas for PI domain."""

from pydantic import BaseModel, Field


class ImportRowIssue(BaseModel):
    """A single problem tied to a CSV file line (header is line 1)."""

    row: int
    message: str


class ImportReport(BaseModel):
    """Outcome of a CSV import: either errors (nothing imported) or counts."""

    ok: bool
    total_rows: int
    imported: int
    skipped: list[ImportRowIssue] = Field(default_factory=list)
    errors: list[ImportRowIssue] = Field(default_factory=list)
